packages/backend-common/src/tokens/AzureAccessToken.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AccessToken,
+  TokenCredential,
+  DefaultAzureCredential,
+} from '@azure/identity';
+import {
+  AzureClientSecretCredential,
+  AzureCredential,
+} from '@backstage/integration';
+import { DateTime } from 'luxon';
+
+/**
+ * Fetches access token information from Azure AD for RDBSM
+ */
+export const fetchAzureAccessToken = async (
+  credential: TokenCredential | DefaultAzureCredential,
+) => {
+  try {
+    const tokenResponse: AccessToken | null = await credential.getToken(
+      'https://ossrdbms-aad.database.windows.net/.default',
+    );
+
+    if (tokenResponse === null) {
+      throw new Error('Azure access token has not been created.');
+    }
+    return {
+      token: tokenResponse.token,
+      expiresOnTimestamp: DateTime.fromMillis(
```

Comment:
Does this function take account of timezones?  I believe the `expriesOnTimestamp` from azure will return a UTC date time, however in the `expirationChecker` it's compared against `DateTime.local()`.